### ISSUE TITLE
[Arm64] CQ - Use CMP (extended register)

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#13867]
```
// TODO-ARM64-CQ: the second register argument of a CMP can be sign/zero
// extended as part of the instruction (using "CMP (extended register)").
// We should use that if possible, swapping operands
// (and reversing the condition) if necessary.
```

**Comment by sdmaclea** [comment]
@jashook Please assign this to me

**Comment by sdmaclea** [comment]
I drafted a change for this, but it was rejected in favor of fixing lowering to not generate mixed type compares. #13941

Implementation of this would depend on a compare being able to contain the cast of op2.

Closing as the issue was largely fixed by #13941 and there is yet to be an indication that further work is necessary.
